pkg/wasm/types.go

```diff
@@ -131,16 +131,16 @@ func (r *RouteRuleConditions) EqualTo(other RouteRuleConditions) bool {
 		}
 	}
 
-	for i := range r.Matches {
-		if !r.Matches[i].EqualTo(other.Matches[i]) {
+	for i := range r.Predicates {
+		if r.Predicates[i] != other.Predicates[i] {
 			return false
 		}
 	}
 
 	return true
 }
 
-type Predicate struct {
+type Condition struct {
```

Comment:
Somewhat confusingly we called the [wasm plugin `Condition`](https://github.com/Kuadrant/wasm-shim/blob/main/src/configuration/action.rs#L16) (i.e. `PatternExpression`) `Predicate`.. but now we have a [`Predicates` field in wasm](https://github.com/Kuadrant/wasm-shim/blob/main/src/configuration/action.rs#L18) and so to retain both I've renamed `Predicate` -> `Condition` and introduced `Predicates` (`[]string`) in routeRuleConditions..